docs/implement-a-customized-actuator-en.md

```diff
@@ -0,0 +1,322 @@
+# Custom SumActuator
+
+Having a tailored-made actuator is key to building a java-tron-based customized public chain. This article illustrates how to develop a java-tron-based `SumActuator`.
+
+Actuator module is divided into 4 different methods that are defined in the `Actuator` interface:
+
+1. `execute`: execute specific actions of transactions, such as state modification, communication between modules, logic execution, etc.
+2. `validate`: define the validation logic of transactions
+3. `getOwnerAddress`: acquire the address of transaction initiators
+4. `calcFee`: define the logic for calculating transaction fees
+
+
+
+## Define and register the contract
+
+Currently, contracts supported by java-tron is defined under src/main/protos/core/contract directory in Protocol module. First creating a math_contract.proto file under this directory and declaring `SumContract`. You can also implement any mathematical calculation you want, such as `Subtraction`.
+
+The logic for `SumContract` is the summation of two numerical values:
+
+```protobuf
+syntax = "proto3";
+package protocol;
+option java_package = "org.tron.protos.contract"; //Specify the name of the package that generated the Java file
+option go_package = "github.com/tronprotocol/grpc-gateway/core";
+message SumContract {
+    int64 param1 = 1;
+    int64 param2 = 2;
+    bytes owner_address = 3;
+}
+```
+
+Meanwhile, register the new contract type in `Transaction.Contract.ContractType` emuneration within the src/main/protos/core/Tron.proto file. Important data structures, such as transactions, accounts and blocks are defined in the Tron.proto file:
+
+```protobuf
+message Transaction {
+  message Contract {
+    enum ContractType {
+      AccountCreateContract = 0;
+      TransferContract = 1;
+      ........
+      SumContract = 52;  
+    }
+  ...
+}
+```
+
+Then register a function to ensure that gRPC can receive and identify the requests of this contracts. Currently, gRPC protocols are all defined in src/main/protos/api/api.proto. Add an `InvokeSum` interface in Wallet Service within api.proto:
+
+```protobuf
+service Wallet {
+  rpc InvokeSum (SumContract) returns (Transaction) {
+    option (google.api.http) = {
+      post: "/wallet/invokesum"
+      body: "*"
+      additional_bindings {
+        get: "/wallet/invokesum"
+      }
+    };
+  };
+  ...
+};
+```
+At last, recompile the modified proto files. directly compiling the java-tron project to compile the proto files is recommended, `protoc` command is also supported.
```

Comment:
proto files => proto file

Reply:
directly compiling the java-tron project to compile the proto files is recommended,
=>
Compiling the java-tron project directly will compile the proto files as well.